src/main/engine/GfxController/headers/GfxController.hpp

```diff
@@ -0,0 +1,115 @@
+/**
+ * @file GfxController.hpp
+ * @author Christian Galvez
+ * @brief 
+ * @version 0.1
+ * @date 2024-01-21
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+#pragma once
+#include <string>
+#include <vector>
+#include <Polygon.hpp>
+
+#define GFX_OK(gType) GfxResult<gType>(GfxApiResult::OK, 0);
+#define GFX_FAILURE(gType) GfxResult<gType>(GfxApiResult::FAILURE, -1);
+
+enum GfxApiResult {
+    OK,
+    FAILURE
+};
+
+enum RenderMode {
+    POINT,
+    LINE,
+    FILL
+};
+
+enum TexFormat {
+    RGBA,
+    RGB,
+    BITMAP
+};
+
+enum TexParam {
+    WRAP_MODE_S,
+    WRAP_MODE_T,
```

Comment:
These seem counter intuitive, but they are important. These enums just map to OpenGL enum values. They're defined here as GfxController enums to fully decouple OpenGL macros from the rest of studious. The assumption is that other graphics backends will behave similar to OpenGL, so these _should_ be used by different backends as well...